### PR TITLE
DSPT3 Unit 2

### PR DESCRIPTION
**Pull request by aidanvu1992**

### CODE REVIEW

**Comment by llpk79** [comment]
### Assignment
- [x] Look at the data. Choose a feature, and plot its relationship with the target.
    - Great work getting a baseline first thing!
    - You're not 'just guessing' here. Using the mean as a baseline is totally legit.
    - Cell converting interest level is broken, look into `Nan` in the column(s).
    - Try some features other than lat and lon!
- [x] Use scikit-learn for linear regression with one feature. You can follow the 5-step process from Jake VanderPlas.
    - Well done.
    - Find your error and compare to that baseline!!
- [x] Define a function to make new predictions and explain the model coefficient.
    - This is sweet, Anh.
- [x] Organize and comment your code.
    - Nice job on this assignment!!
### Stretch Goals
- [ ] Do linear regression with two or more features.
    - Please?

**Comment by llpk79** [comment]
## Regression 2
- [x] Do train/test split. Use data from April & May 2016 to train. Use data from June 2016 to test.
- [x] Engineer at least two new features. (See below for explanation & ideas.)
- [x] Fit a linear regression model with at least two features.
- [x] Get the model's coefficients and intercept.
- [x] Get regression metrics RMSE, MAE, and $R^2$, for both the train and test data.
    - Our baseline is the error when using the mean as our guess, not the mean itself.
- [x] What's the best test MAE you can get? Share your score and features used with your cohort on Slack!
    - Great function! Try some different combinations of features!
- [x] As always, commit your notebook to your fork of the GitHub repo.
    - ✔️ 

Nice work, Anh!

**Comment by llpk79** [comment]
## Ridge Regression
- [x] Do train/test split. Use data from January — March 2019 to train. Use data from April 2019 to test.
    - Nicely done.
- [x] Do one-hot encoding of categorical features.
    - Kinda strange that these `*_SQUARE_FEET` columns are categorical...
- [x] Do feature selection with `SelectKBest`.
    - Excellent work automating the search for the best `k`!
    - How can you store and report the best result?
    - How can you get rid of all of those warnings? lol
- [x] Do [feature scaling](https://scikit-learn.org/stable/modules/preprocessing.html).
- [x] Fit a ridge regression model with multiple features.
- [x] Get mean absolute error for the test set.
- [x] As always, commit your notebook to your fork of the GitHub repo.
    - ✔️ 

Great job, Ahn!

**Comment by llpk79** [comment]
## Logistic Regression
- [x] Do train/validate/test split. Train on reviews from 2016 & earlier. Validate on 2017. Test on 2018 & later.
- [x] Begin with baselines for classification.
    - Nice work checking baseline for validation set.
- [x] Use scikit-learn for logistic regression.
- [x] Get your model's validation accuracy. (Multiple times if you try multiple iterations.)
- [x] Get your model's test accuracy. (One time, at the end.)
    - All clean and concise.
- [x] Commit your notebook to your fork of the GitHub repo.
    - ✔️ 

## Stretch Goals
- [ ] Add your own stretch goal(s) !
- [ ] Make exploratory visualizations.
- [x] Do one-hot encoding.
- [x] Do [feature scaling](https://scikit-learn.org/stable/modules/preprocessing.html).
- [ ] Get and plot your coefficients.
- [ ] Try [scikit-learn pipelines](https://scikit-learn.org/stable/modules/compose.html).

Realy nice, Anh!
